### PR TITLE
Feat/add mm e2e mmi build

### PR DESCRIPTION
**Pull request by racitores**
## Explanation

Currently MMI extension is assuming that all general/core MM extension features should work under MMI but this is not been tested in the pipeline.

This PR adds Metamask e2e tests run in the circle ci pipeline for a MMI build generated extension. There are a set failing tests marked as `@no-mmi` so that when running the tests with `yarn test:e2e:chrome:mmi` will do `SELENIUM_BROWSER=chrome node test/e2e/run-all.js --mmi` passing --mmi as param.

This is a first step to improve MMI test coverage and regression. Next steps will be reviewed and updated tests if required

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by zone-live — DISMISSED** [review]
💪🏼

**Review by brad-decker — CHANGES_REQUESTED** [review]
@PeterYinusa you should probably review this and see if we should include some way of documenting the `@no-mmi` flag in our E2E documents. Blocking for review but overall the code looks good for me.

File: test/e2e/run-e2e-test.js [hunk 1]
```diff
@@ -117,6 +122,10 @@ async function main() {
   const configFile = path.join(__dirname, '.mocharc.js');
   const extraArgs = process.env.E2E_ARGS?.split(' ') || [];
 
+  if (mmi) {
+    extraArgs.push('-g', '@no-mmi', '-i');
+  }
+
```
Comment (brad-decker) [inline test/e2e/run-e2e-test.js:128]: Could you add inline documentation here for what these flags do (grep/invert) and explain the @no-mmi tag on some of the tests

**Comment by PeterYinusa** [comment]
> @PeterYinusa you should probably review this and see if we should include some way of documenting the `@no-mmi` flag in our E2E documents. Blocking for review but overall the code looks good for me.

Thanks for the heads up @brad-decker, this seems in line with the approach discussed with @racitores.
It looks good to me, but I agree we should add some documentation around the usage of the tags to ensure contributors understand when to use them.

**Review by PeterYinusa — DISMISSED** [review]
Looks great, please add documentation for contributors

**Review by zone-live — APPROVED** [review]

**Review by PeterYinusa — APPROVED** [review]
🚀

**Review by brad-decker — APPROVED** [review]

**Pull request merged**